ccl/serverccl: TestTenantVars failed

ccl/serverccl.TestTenantVars [failed](https://tanzanite.cluster.engflow.com/invocations/default/01563976-1fe2-4191-bba5-22eea9550bd0?testReportRun=4&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NlcnZlcmNjbDpzZXJ2ZXJjY2xfdGVzdA==) on master @ [0b7ae19e2b94b851ed8812914f57032aab699811](https://github.com/cockroachdb/cockroach/commits/0b7ae19e2b94b851ed8812914f57032aab699811):


```
=== RUN   TestTenantVars
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantVars411347686
    test_log_scope.go:81: use -show-logs to present logs inline
    tenant_vars_test.go:153: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantVars411347686
--- FAIL: TestTenantVars (1.33s)
=== RUN   TestTenantVars/shared-process=true
    tenant_vars_test.go:143: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/serverccl/tenant_vars_test.go:143
        	            				github.com/cockroachdb/cockroach/pkg/testutils/subtest.go:28
        	Error:      	"1e+07" is not less than or equal to "0"
        	Test:       	TestTenantVars/shared-process=true
    --- FAIL: TestTenantVars/shared-process=true (0.37s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/multi-tenant @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantVars.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36151

ccl/serverccl.TestTenantVars [failed](https://tanzanite.cluster.engflow.com/invocations/default/7f1d73c5-5ddd-4855-bf60-ae0b8b268601?testReportRun=22&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NlcnZlcmNjbDpzZXJ2ZXJjY2xfdGVzdA==) on master @ [e50b0ec4d3a53f81e26f3776ae3f3be55d435a9a](https://github.com/cockroachdb/cockroach/commits/e50b0ec4d3a53f81e26f3776ae3f3be55d435a9a):


```
=== RUN   TestTenantVars
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantVars1514036488
    test_log_scope.go:81: use -show-logs to present logs inline
    tenant_vars_test.go:153: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantVars1514036488
--- FAIL: TestTenantVars (1.33s)
=== RUN   TestTenantVars/shared-process=false
    tenant_vars_test.go:143: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/serverccl/tenant_vars_test.go:143
        	            				github.com/cockroachdb/cockroach/pkg/testutils/subtest.go:28
        	Error:      	"1e+07" is not less than or equal to "0"
        	Test:       	TestTenantVars/shared-process=false
    --- FAIL: TestTenantVars/shared-process=false (0.48s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=22</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantVars.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/serverccl.TestTenantVars [failed](https://tanzanite.cluster.engflow.com/invocations/default/b1026e4d-beeb-490a-89a4-c9d5cb0a6c97?testReportRun=26&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NlcnZlcmNjbDpzZXJ2ZXJjY2xfdGVzdA==) on master @ [34caf9d2c267f95f6aef2713708806e6f14948d6](https://github.com/cockroachdb/cockroach/commits/34caf9d2c267f95f6aef2713708806e6f14948d6):


```
=== RUN   TestTenantVars
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantVars3081158596
    test_log_scope.go:81: use -show-logs to present logs inline
    tenant_vars_test.go:153: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantVars3081158596
--- FAIL: TestTenantVars (1.35s)
=== RUN   TestTenantVars/shared-process=true
    tenant_vars_test.go:143: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/serverccl/tenant_vars_test.go:143
        	            				github.com/cockroachdb/cockroach/pkg/testutils/subtest.go:28
        	Error:      	"1e+07" is not less than or equal to "0"
        	Test:       	TestTenantVars/shared-process=true
    --- FAIL: TestTenantVars/shared-process=true (0.36s)
```

Parameters:
 - <code>=true</code>
 - <code>attempt=1</code>
 - <code>run=26</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantVars.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/serverccl.TestTenantVars [failed](https://tanzanite.cluster.engflow.com/invocations/default/efc8a553-d34b-4b19-ae9c-5efa8025afc8?testReportRun=24&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NlcnZlcmNjbDpzZXJ2ZXJjY2xfdGVzdA==) on master @ [f0116ea373a2b87155e7f0264df4f783ce177360](https://github.com/cockroachdb/cockroach/commits/f0116ea373a2b87155e7f0264df4f783ce177360):


```
=== RUN   TestTenantVars
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantVars1251763533
    test_log_scope.go:81: use -show-logs to present logs inline
    tenant_vars_test.go:153: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantVars1251763533
--- FAIL: TestTenantVars (1.39s)
=== RUN   TestTenantVars/shared-process=true
    tenant_vars_test.go:143: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/serverccl/tenant_vars_test.go:143
        	            				github.com/cockroachdb/cockroach/pkg/testutils/subtest.go:28
        	Error:      	"1e+07" is not less than or equal to "0"
        	Test:       	TestTenantVars/shared-process=true
    --- FAIL: TestTenantVars/shared-process=true (0.44s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=24</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantVars.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/serverccl.TestTenantVars [failed](https://tanzanite.cluster.engflow.com/invocations/default/4fd5bdf6-3b60-4551-ad0a-0177ca687dc0?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NlcnZlcmNjbDpzZXJ2ZXJjY2xfdGVzdA==) on master @ [31e84cb3a57c52a779ff0982c95fb26646b54926](https://github.com/cockroachdb/cockroach/commits/31e84cb3a57c52a779ff0982c95fb26646b54926):


```
=== RUN   TestTenantVars
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestTenantVars889137562
    test_log_scope.go:76: use -show-logs to present logs inline
    tenant_vars_test.go:151: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantVars889137562
--- FAIL: TestTenantVars (8.81s)
=== RUN   TestTenantVars/shared-process=true
    tenant_vars_test.go:139: 
        	Error Trace:	pkg/ccl/serverccl/tenant_vars_test.go:139
        	            				pkg/testutils/subtest.go:23
        	Error:      	"3e+07" is not less than or equal to "2e+07"
        	Test:       	TestTenantVars/shared-process=true
    --- FAIL: TestTenantVars/shared-process=true (1.49s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantVars.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/serverccl.TestTenantVars [failed](https://tanzanite.cluster.engflow.com/invocations/default/c2e434f4-06dc-4674-8103-33fda8c7137b?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NlcnZlcmNjbDpzZXJ2ZXJjY2xfdGVzdA==) on master @ [358c9a5ada822ab2cdc0aa09baa2f08028fadc3c](https://github.com/cockroachdb/cockroach/commits/358c9a5ada822ab2cdc0aa09baa2f08028fadc3c):


```
=== RUN   TestTenantVars
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestTenantVars3177131538
    test_log_scope.go:76: use -show-logs to present logs inline
    tenant_vars_test.go:151: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantVars3177131538
--- FAIL: TestTenantVars (4.03s)
=== RUN   TestTenantVars/shared-process=false
    tenant_vars_test.go:139: 
        	Error Trace:	pkg/ccl/serverccl/tenant_vars_test.go:139
        	            				pkg/testutils/subtest.go:23
        	Error:      	"1e+07" is not less than or equal to "0"
        	Test:       	TestTenantVars/shared-process=false
    --- FAIL: TestTenantVars/shared-process=false (0.56s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantVars.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/serverccl/tenant_vars_test.go
+++ b/pkg/ccl/serverccl/tenant_vars_test.go
@@ -69,17 +69,17 @@ func TestTenantVars(t *testing.T) {
 		metrics, err := parser.TextToMetricFamilies(resp.Body)
 		require.NoError(t, err)
 
-		userCPU, found := metrics["sys_cpu_user_ns"]
+		tenantUserCPU, found := metrics["sys_cpu_user_ns"]
 		require.True(t, found)
-		require.Len(t, userCPU.GetMetric(), 1)
-		require.Equal(t, io_prometheus_client.MetricType_GAUGE, userCPU.GetType())
-		cpuUserNanos := userCPU.Metric[0].GetGauge().GetValue()
+		require.Len(t, tenantUserCPU.GetMetric(), 1)
+		require.Equal(t, io_prometheus_client.MetricType_GAUGE, tenantUserCPU.GetType())
+		tenantCpuUserNanos := tenantUserCPU.Metric[0].GetGauge().GetValue()
 
-		sysCPU, found := metrics["sys_cpu_sys_ns"]
+		tenantSysCPU, found := metrics["sys_cpu_sys_ns"]
 		require.True(t, found)
-		require.Len(t, sysCPU.GetMetric(), 1)
-		require.Equal(t, io_prometheus_client.MetricType_GAUGE, sysCPU.GetType())
-		cpuSysNanos := sysCPU.Metric[0].GetGauge().GetValue()
+		require.Len(t, tenantSysCPU.GetMetric(), 1)
+		require.Equal(t, io_prometheus_client.MetricType_GAUGE, tenantSysCPU.GetType())
+		tenantCpuSysNanos := tenantSysCPU.Metric[0].GetGauge().GetValue()
 
 		now, found := metrics["sys_cpu_now_ns"]
 		require.True(t, found)
@@ -87,23 +87,22 @@ func TestTenantVars(t *testing.T) {
 		require.Equal(t, io_prometheus_client.MetricType_GAUGE, now.GetType())
 		nowNanos := now.Metric[0].GetGauge().GetValue()
 
-		uptime, found := metrics["sys_uptime"]
+		tenantUptime, found := metrics["sys_uptime"]
 		require.True(t, found)
-		require.Len(t, uptime.GetMetric(), 1)
-		require.Equal(t, io_prometheus_client.MetricType_GAUGE, uptime.GetType())
-		uptimeSeconds := uptime.Metric[0].GetGauge().GetValue()
+		require.Len(t, tenantUptime.GetMetric(), 1)
+		require.Equal(t, io_prometheus_client.MetricType_GAUGE, tenantUptime.GetType())
+		uptimeSeconds := tenantUptime.Metric[0].GetGauge().GetValue()
 
 		// The values are between zero and whatever User/Sys time is observed after the get.
 		require.LessOrEqual(t, float64(startNowNanos), nowNanos)
 		require.LessOrEqual(t, nowNanos, float64(timeutil.Now().UnixNano()))
 
-		cpuTime := gosigar.ProcTime{}
-		require.NoError(t, cpuTime.Get(os.Getpid()))
-		require.LessOrEqual(t, 0., cpuUserNanos)
-		require.LessOrEqual(t, cpuUserNanos, float64(cpuTime.User)*1e6)
-		require.LessOrEqual(t, 0., cpuSysNanos)
-		require.LessOrEqual(t, cpuSysNanos, float64(cpuTime.Sys)*1e6)
-
+		testCpuTime := gosigar.ProcTime{}
+		require.NoError(t, testCpuTime.Get(os.Getpid()))
+		require.LessOrEqual(t, 0., tenantCpuUserNanos)
+		require.LessOrEqual(t, tenantCpuUserNanos, float64(testCpuTime.User)*1e6)
+		require.LessOrEqual(t, 0., tenantCpuSysNanos)
+		require.LessOrEqual(t, tenantCpuSysNanos, float64(testCpuTime.Sys)*1e6)
 		require.LessOrEqual(t, 0., uptimeSeconds)
 
 		resp, err = client.Get(url)
@@ -115,30 +114,29 @@ func TestTenantVars(t *testing.T) {
 		metrics, err = parser.TextToMetricFamilies(resp.Body)
 		require.NoError(t, err)
 
-		userCPU, found = metrics["sys_cpu_user_ns"]
+		tenantUserCPU, found = metrics["sys_cpu_user_ns"]
 		require.True(t, found)
-		require.Len(t, userCPU.GetMetric(), 1)
-		require.Equal(t, io_prometheus_client.MetricType_GAUGE, userCPU.GetType())
-		cpuUserNanos2 := userCPU.Metric[0].GetGauge().GetValue()
+		require.Len(t, tenantUserCPU.GetMetric(), 1)
+		require.Equal(t, io_prometheus_client.MetricType_GAUGE, tenantUserCPU.GetType())
+		tenantCpuUserNanos2 := tenantUserCPU.Metric[0].GetGauge().GetValue()
 
-		sysCPU, found = metrics["sys_cpu_sys_ns"]
+		tenantSysCPU, found = metrics["sys_cpu_sys_ns"]
 		require.True(t, found)
-		require.Len(t, sysCPU.GetMetric(), 1)
-		require.Equal(t, io_prometheus_client.MetricType_GAUGE, sysCPU.GetType())
-		// cpuSysNanos2 := sysCPU.Metric[0].GetGauge().GetValue()
+		require.Len(t, tenantSysCPU.GetMetric(), 1)
+		require.Equal(t, io_prometheus_client.MetricType_GAUGE, tenantSysCPU.GetType())
+		tenantCpuSysNanos2 := tenantSysCPU.Metric[0].GetGauge().GetValue()
 
-		uptime, found = metrics["sys_uptime"]
+		tenantUptime, found = metrics["sys_uptime"]
 		require.True(t, found)
-		require.Len(t, uptime.GetMetric(), 1)
-		require.Equal(t, io_prometheus_client.MetricType_GAUGE, uptime.GetType())
-		uptimeSeconds2 := uptime.Metric[0].GetGauge().GetValue()
+		require.Len(t, tenantUptime.GetMetric(), 1)
+		require.Equal(t, io_prometheus_client.MetricType_GAUGE, tenantUptime.GetType())
+		uptimeSeconds2 := tenantUptime.Metric[0].GetGauge().GetValue()
 
 		cpuTime2 := gosigar.ProcTime{}
 		require.NoError(t, cpuTime2.Get(os.Getpid()))
 
-		require.LessOrEqual(t, float64(cpuTime2.User-cpuTime.User)*1e6, cpuUserNanos2)
-		// TODO(#119329): Sometimes our metrics have 0 cpuSysNanos.
-		// require.LessOrEqual(t, float64(cpuTime2.Sys-cpuTime.Sys)*1e6, cpuSysNanos2)
+		require.Less(t, tenantCpuUserNanos2, float64(cpuTime2.User)*1e6)
+		require.LessOrEqual(t, tenantCpuSysNanos2, float64(cpuTime2.Sys)*1e6)
 		require.LessOrEqual(t, uptimeSeconds, uptimeSeconds2)
 
 		_, found = metrics["jobs_running_non_idle"]

